src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt

```diff
@@ -24,13 +27,54 @@ import com.google.protobuf.ByteString
 import com.google.pubsub.v1.PubsubMessage
 import com.google.pubsub.v1.TopicName
 import org.slf4j.LoggerFactory
+import javassist.CtMethod.ConstParameter.string
 
-fun getResultData(taskId: String, compiledBytes: ByteString, outputFileName: String, statusCode: Int): ByteString {
-    return CompilerFleet.CompilerFleetResult.newBuilder()
+
+
+fun buildResultData(taskId: String, compiledBytes: ByteString, statusCode: Int): ByteString {
+    val status = when (statusCode) {
+        0 -> CompileResponse.Status.OK
+        else -> CompileResponse.Status.FAILED
+    }
+
+    val texbeResponse = CompileResponse.newBuilder()
+            .setPdfFile(compiledBytes)
+            .setStatus(status)
+            .build()
+
+    val requestData = CompilerFleet.RequestIdentifier.newBuilder()
```

Comment:
if you construct identifier then you may want to name the variable appropriately